### PR TITLE
gitkraken: add darwin support

### PR DESCRIPTION
**Pull request by arkivm**
###### Motivation for this change
gitkraken: Add support for darwin platforms

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arkivm** [comment]
@evanjs

**Review by evanjs — APPROVED** [review]
```
result ⇒ /nix/store/q176zbj8ffrbd9vc2dv4xwydl8hwg286-gitkraken-8.1.0
```
Screen Shot 2021-11-03 at 11 00 59 PM

`x86_64-darwin`: ✅ 

Thank you!

**Comment by arkivm** [comment]
@SuperSandro2000 can we merge this?

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 1]
```diff
@@ -3,114 +3,142 @@
...
+      runHook preInstall
+
+      mkdir -p $out/share/gitkraken/
+      cp -R $src/* $out/share/gitkraken/
+
+      mkdir -p $out/bin
+      ln -s $out/share/gitkraken/gitkraken $out/bin/gitkraken
+
+      mkdir -p $out/share/applications
+      ln -s ${desktopItem}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gitkraken/default.nix:112]: Please use copyDekstopItem.

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 2]
```diff
@@ -3,114 +3,142 @@
...
-  postFixup = ''
-    pushd $out/share/gitkraken
-    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" gitkraken
-
-    for file in $(find . -type f \( -name \*.node -o -name gitkraken -o -name \*.so\* \) ); do
-      patchelf --set-rpath ${libPath}:$out/share/gitkraken $file || true
-    done
-    popd
-  '';
+  src = srcs.${stdenv.hostPlatform.system};
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gitkraken/default.nix:31]: This is missing a throw unsupported platform.

**Comment by arkivm** [comment]
#67867 should be closed now

**Pull request merged**